### PR TITLE
[3.0] Return false when attempting to delete a non-existent directory

### PR DESCRIPTION
**Pull request by phil-davis**
Backport PR #1848 to `3.0` branch.

Issue #1847

### CODE REVIEW

**Comment by jack-worman** [comment]
Is this bug relatively new, wouldn't want to break code relying on this "feature".

**Comment by phil-davis** [comment]
See discussion in #1848 

This PR is just a backport.

File: phpseclib/Net/SFTP.php [hunk 1]
```diff
@@ -1100,8 +1114,9 @@ private function readlist($dir, $raw = true)
                 break;
             case NET_SFTP_STATUS:
                 // presumably SSH_FX_NO_SUCH_FILE or SSH_FX_PERMISSION_DENIED
-                $this->logError($response);
-                return false;
+                [$status] = Strings::unpackSSH2('N', $response);
```
Comment (jack-worman) [inline phpseclib/Net/SFTP.php:1117]: You must use `list()`

**Comment by jack-worman** [comment]
I understand it is a backport. But now that we are targeting 3.0, we must consider backwards compatibility.

I believe this bug has been in a release branch for about a month, which is not too bad. Ultimately its up to @terrafrost

**Comment by terrafrost** [comment]
@jack-worman - yah I'm good with it.

@phil-davis - are you gonna backport this to 1.0 and 2.0 as well? If not that's cool - I'll backport it myself if you don't want to!

**Pull request closed without merging**